hal/lib/athena/handles/HandlesInternal.h

```diff
@@ -0,0 +1,31 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) FIRST 2016. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <stdint.h>
+
+#include "HAL/Handles.h"
+
+namespace hal {
+static inline int16_t getHandleIndex(HalHandle handle) {
+  return (int16_t)(handle & 0xffff);
+}
+static inline HalHandleEnum getHandleType(HalHandle handle) {
+  return (HalHandleEnum)((handle >> 24) & 0xff);
```

Comment:
It isn't immediately obvious why the shifts and masks are happening the way they are. A data layout description somewhere would help.